src/lib/validationRules/remoteDocumentAsJsonLd/remoteDocumentAsJsonLd.ts

```diff
@@ -0,0 +1,191 @@
+          title: `Client Identifier Document could not be fetched`,
+          description: `The URI \`${context.documentIri}\` could not be resolved:\n${error}`,
+          affectedFields: [
+            { fieldName: "client_id", fieldValue: context.documentIri },
+          ],
+        },
+      ];
+    }
+
+    if (fetchResult.redirected) {
+      return [
+        {
+          status: "warning",
+          title: "Unexpected redirect",
+          description:
+            "Upon fetching Client Identifier Document, redirect happened",
+          affectedFields: [
+            { fieldName: "client_id", fieldValue: context.documentIri },
+          ],
+        },
+      ];
+    }
+
+    if (fetchResult.status !== 200) {
+      return [
+        {
+          status: "error",
+          title: "Unexpected status code",
+          description: `The status code from fetching Client Identifier Document was ${fetchResult.status} but should have been 200.`,
+          affectedFields: [
+            { fieldName: "client_id", fieldValue: context.documentIri },
+          ],
+        },
+      ];
+    }
+
+    const results: RuleResult[] = [];
+
+    let remoteDocument: ClientIdDocument;
+    try {
+      // we need this to allow error-less passing from .json() to ClientIdDocument
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      remoteDocument = (await fetchResult.json()) as any;
+    } catch (error) {
+      results.push({
+        status: "error",
+        title: "Remote Client Identifier Document could not be parsed",
+        description:
+          "The resource located at the Client Identifier URI is not a valid JSON document.",
+        affectedFields: [
+          { fieldName: "client_id", fieldValue: context.documentIri },
+        ],
+      });
+      return results;
+    }
+
+    const contentType = fetchResult.headers.get("content-type");
+    const mimeType = !contentType ? "" : contentType.replace(/;.*/, "");
+
+    if (mimeType !== "application/json" && mimeType !== "application/ld+json") {
+      results.push({
+        status: "error",
+        title: "Invalid `content-type` header fetching Client Identifier",
+        description: `The Response Header \`content-type\` must have a MIME type of \`application/ld+json\` or \`application/json\` but was \`${mimeType}\``,
+        affectedFields: [
+          { fieldName: "content-type", fieldValue: contentType },
+        ],
+      });
+    }
+
+    if (!remoteDocument["@context"]) {
+      results.push({
+        status: "error",
+        title: "Remote Client Identifier Document misses `@context` field",
+        description: `The remote Client Identifier Document has no @context field set but it should be set to \`${OIDC_CONTEXT}\`.`,
+        affectedFields: [
+          { fieldName: "content-type", fieldValue: contentType },
+        ],
+      });
+      return results;
+    }
+
+    if (
+      remoteDocument["@context"] !== OIDC_CONTEXT &&
```

Comment:
I'm not sure but I think this is misunderstanding JSON-LD, both these are equivalent:

```
{ "@context": "https://context.example" }
```
```
{ "@context": [ "https://context.example" ] }